RFP-01.md

```diff
@@ -0,0 +1,212 @@
+     */
+    function setMinCow(uint256 cowAmt) external auth;
+    /**
+     * Bill a sub-pool and send the proceeds to the caller
+     * The `auth` is unable to bill the sub-pool if the sub-pool has announced an exit
+     * and the `delay` has elapsed.
+     * @param pool Which sub-pool to bill
+     * @param tokens The tokens to bill
+     * @param amts The amount of tokens to bill
+     * @param reason The reason for the bill (fine, etc)
+     * @dev Emits an event `Bill` when billing a sub-pool
+     */
+    function bill(SubPool pool, IERC20[] calldata tokens, uint256[] calldata amts, string calldata reason) external auth;
+
+    // --- SubPool authed functions ---
+
+    /**
+     * Indicate a sub-pool's intent to exit the batch auction
+     * At this point the solver whose sub-pool it is is now removed from
+     * the competition.
+     * This then starts the `delay` timer until exit is allowed.
+     * @param pool The sub-pool to announce an exit for
+     * @dev Emits an event `AnnounceExit`
+     */
+    function announceExit(SubPool pool) external onlySubpoolAuth;
+    /**
+     * Withdraw any remaining collateral / cow from the sub-pool.
+     * Funds are sent to `msg.sender`.
+     * @dev Will revert if the sub-pool has not announced the exit and/or the
+     * `delay` has not passed.
+     * @dev MUST emit an event `Exit` on the factory
+     * @dev This function can be called repeatedly so long as the sub-pool has
+     * announced an exit and the `delay` has passed. This allows for the
+     * withdrawal of any remaining collateral / cow (or that which was accidentally
+     * sent to the sub-pool).
+     * @param pool The sub-pool whose exit to finalize
+     * @param tokens The tokens to withdraw
+     */
+    function exit(SubPool pool, IERC20[] calldata tokens) external onlySubpoolAuth;
+    /**
+     * Set the backend URI for the sub-pool
+     * @param pool The sub-pool to set the URI for
+     * @param newUri The new URI to set
+     * @dev Emits an event `SetBackendUri` on the factory when setting a URI
+     */
+    function setBackendUri(SubPool pool, string calldata newUri) external onlySubpoolAuth;
+}
+
+interface SignedSettlement {
+    /**
+     * Signed settlement function that is callable by `msg.sender` if `msg.sender` is in
+     * receipt of a valid signature. The signer is a trusted entity and bears the responsibility
+     * of ensuring that `msg.sender` is correctly collateralised.
+     *  
+     * The signed data **MUST** include:
+     * - `solver` - the address of the sub-bonded solver, i.e. `msg.sender`
+     * - `deadline` - the deadline by which the settlement must be executed (e.g. block number)
+     * - `tokens` - the traded tokens vector for the settlement
+     * - `clearingPrices` - the clearing prices vector for the settlement
+     * - `trades` - the trades vector for the settlement
+     * - `interactions` - the interactions vector for the settlement
```

Comment:
Do we want to mention the fact that optionally we'd like some of the interactions to be optional or not?

Reply:
The context for this is intentionally left open, as I expect both options to be analysed with a concrete gas difference so that:
1. There is quantitative analysis on what it would cost to do optional interactions, vs not.
2. This can then be put to backend, in the event that optional interactions is more expensive, for a decision as to whether or not this is fundamentally worth it.

Reply:
There is no context anywhere in this document that says we're interested in only signing some of the interactions (which is almost a business requirement according to [this](https://github.com/cowdao-grants/rfps/pull/1#discussion_r1670008722)).
I'd expect a grantee to just sign what we say needs to be signed. At most, an enterprising grantee would suggest to trim out all interactions and deal with the risks for cheaper execution. But I really don't see why anyone would suggest only signing some: it's clearly much more work and is likely to be significantly more complex or a lot more gas expensive.